avm/res/cdn/profile/customdomain/main.bicep

```diff
@@ -0,0 +1,77 @@
+metadata name = 'CDN Profiles Custom Domains'
+metadata description = 'This module deploys a CDN Profile Custom Domains.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the custom domain.')
+param name string
+
+@description('Required. The name of the CDN profile.')
+param profileName string
+
+@description('Required. The host name of the domain. Must be a domain name.')
+param hostName string
+
+@description('Optonal. Resource reference to the Azure DNS zone.')
+param azureDnsZoneResourceId string = ''
+
+@description('Optional. Key-Value pair representing migration properties for domains.')
+param extendedProperties object = {}
+
+@description('Optional. Resource reference to the Azure resource where custom domain ownership was prevalidated.')
+param preValidatedCustomDomainResourceId string = ''
+
+@allowed([
+  'CustomerCertificate'
+  'ManagedCertificate'
+])
+@description('Required. The type of the certificate used for secure delivery.')
+param certificateType string
+
+@allowed([
+  'TLS10'
+  'TLS12'
+])
+@description('Optional. The minimum TLS version required for the custom domain. Default value: TLS12.')
+param minimumTlsVersion string = 'TLS12'
+
+@description('Optional. The name of the secret. ie. subs/rg/profile/secret.')
+param secretName string = ''
+
+resource profile 'Microsoft.Cdn/profiles@2023-05-01' existing = {
+  name: profileName
+
+  resource profile_secrect 'secrets@2023-05-01' existing = if (!empty(secretName)) {
+    name: secretName
+  }
+}
+
+resource profile_custom_domain 'Microsoft.Cdn/profiles/customDomains@2023-05-01' = {
```

Comment:
Should be `customDomain`

Reply:
ok: https://github.com/Azure/bicep-registry-modules/pull/1106/commits/264d035f36513fe6c268ee0c65fc01340ca5eacd